dataflow/src/org/checkerframework/dataflow/analysis/Analysis.java

```diff
@@ -462,12 +475,24 @@ protected void addStoreBefore(
             Block b, Node node, S s, Store.Kind kind, boolean addBlockToWorklist) {
         S thenStore = getStoreBefore(b, Store.Kind.THEN);
         S elseStore = getStoreBefore(b, Store.Kind.ELSE);
+        boolean shouldWiden = false;
+        Integer count = null;
+        if (blockCount != null) {
+            count = blockCount.get(b);
+            count = count == null ? 0 : count;
+            shouldWiden = count >= maxCountBeforeWidening;
+        }
 
         switch (kind) {
             case THEN:
                 {
                     // Update the then store
-                    S newThenStore = (thenStore != null) ? thenStore.leastUpperBound(s) : s;
+                    S newThenStore;
```

Comment:
This logic is duplicated 5 times. Extract in helper method.
I would check for `(thenStore == null)` just once first, as that is common to the two branches.